### ISSUE TITLE
md-input causes v-model lag on mobile

### ISSUE THREAD

**Issue by rayfoss** [vuematerial/vue-material#1835]
### Version
vue 2.5.16
vue-material 1.0.0-beta-10.2

### Reproduction link
[https://codesandbox.io/s/xk52mqm7o](https://codesandbox.io/s/xk52mqm7o)

### Steps to reproduce
type two words on Chrome 66  on an Android phone or emulator. The pixel and pixel images work.

### What is expected?
When you click "create todo" validateForm reads form.todo. Unfortunately, it only updates after every space bar press.

### What is actually happening?
v-model contains everything but the last word

---
I tried v-model.lazy and initializing form to `form: todo: ''`... same issue

![kapture 2018-06-28 at 13 54 57](https://user-images.githubusercontent.com/876076/42055778-bece1a16-7add-11e8-91cf-2a325bb4a7e4.gif)

**Comment by rayfoss** [comment]
@Samuell1 not surprising... disabling spell check makes the problem unnoticeable. This is a race condition. 

My first guess is @click is processed before nextTick, but that doesn't explain why the `<input>` works. <MaterialInput> brute force fixes the issue. Any ideas what I should be looking at? My first inclination is to try to get rid of the watch's and try to use the property more directly

**Comment by rayfoss** [comment]
Good news and bad news:
![screen shot 2018-06-29 at 2 55 34 pm](https://user-images.githubusercontent.com/876076/42112154-98006aea-7bac-11e8-89df-0ba45a3967d7.png)
![screen shot 2018-06-29 at 2 55 25 pm](https://user-images.githubusercontent.com/876076/42112155-98194894-7bac-11e8-9af0-b4af407eb5fc.png)

**Comment by Samuell1** [comment]
@rayfoss can you define your v-model in data? because form.todo doesnt not exists in your object, that is maybe causing issue

**Comment by rayfoss** [comment]
@Samuell1 Yes, I tried that... turns out the setter for the form will handle simple values without the need for an additional per value setter, at least on 2.5.

The best solution I've found so far is to do change the MdInput template to 
```
<template>
  <input
    class="md-input"
    v-model="model"
    v-bind="attributes"
    v-on="listeners"
    @input="$emit('input', $event.target.value)"
    @focus="onFocus"
    @blur="onBlur">
</template>
```

The downside is that MdField also emits changes on nextTick... so on desktop that would be twice the events as there are now. Setting the MdFieldMixin model in data instead of a computer setter will get it updated faster, but that won't do us any good as we still need to emit somehow. I tried emitting from the setter, that won't reach us in time.

I think the best solution is to get rid of MdFieldMixin emit's and make the individual components do the heavy lifting where possible. If not possible, take a "slow" property to make it clear you'll need to wait till nextTick to use the data.

**Comment by rayfoss** [comment]
Update: alternative solution
looks like the computed model setter does get the message in time, and an emit from there will find it's way to the parent in time. But the race condition is still a problem. There are still several watches around that won't react till nextTick, and they'll set the model back to the original value.

```
computed: {
    model: {
      get () {
        return this.localValue
      },
      set (value) {
        this.$emit('input', value)
        console.log(value)
        this.localValue = value
      }
    },
```

Question remains... why are we currently waiting till nextTick anyway? @Samuell1
What does this do? why wait till next tick, allowing for the race condition to exist... why are we waiting to emit, why not emit here?
```
  if (value.constructor.toString().match(/function (\w*)/)[1].toLowerCase() !== 'inputevent') {
          this.$nextTick(() => {
            this.localValue = value
          })
        }
```

**Comment by rayfoss** [comment]
Well I didn't break any tests so it must not be important, jk. Maybe were over thinking this... It's clearly to make all.changes happen on the same tick, as there are a lot of slow watches. 

The purpose of the regex is not important to us for this issue. We just needed to update fater, which the PR focuses on.

**Comment by mismith** [comment]
Just released a Cordova app to the wild and this is causing the app to crash on even modern iOS devices.. :/

**Comment by rayfoss** [comment]
@mismith while you shouldn't have to do this, it's probably a good idea to automatically require all submit handlers to have a next tick wrapper in them... no idea how to automate the enforcement of that.

**Comment by mismith** [comment]
In my case I'm not even using one, I'm filtering a list of results (after minor debouncing), so that won't fix much for me.

**Comment by rayfoss** [comment]
@Samuell1 this is very very much a confirmed bug that haunts me daily, can we add the `bug` and `Pull-Request-available` label to this

**Comment by rayfoss** [comment]
@marcosmoura any plans to fix this? I can pitch in more as well. I run into this bug almost daily. See my PR... if you'd like me to refactor it for another branch, I can

**Comment by agr1277** [comment]
I am also experiencing this behavior, and it is causing severe issues. In fact, I have at least one instance where I have to wrap the submit handler in two $nextTick callbacks before it picks up the input. If @rayfoss's solution is valid, I'd really appreciate it being merged in.
